src/res_tracker.cc

```diff
@@ -44,13 +45,17 @@ void ResTracker::Extract(ResTracker* removed) {
     return;
   }
 
-  parent1_ = res_->state_id();
-  parent2_ = 0;
+  if (res_->quantity() > eps_rsrc()) {
+    parent1_ = res_->state_id();
+    parent2_ = 0;
+    
+    Record();
+  }
```

Comment:
@nuclearkatie Now that we're digging into when we Record things, and my understanding is expanding... I'm a little concerned about this check for size > `eps_rsrc()`.  Don't we want to record that this resource is depleted?  It should certainly have it's state ID bumped and be in a different state now than it was before.  Is that happening reliably somwhere else?

Reply:
No, this is not happening elsewhere, it is not typically recorded when resources cease to have any quantity. 
From what I understand, outside of packaging, Absorb is the main way that happens. 

Absorb does not bump the absorbed resource, as a second new, zero resource. It only creates one new resource with parent1 resource and parent2 absorbed resource. I personally do not think it would make sense for the logic to flow:
```Resource + Asborbed => NewBigResource with two parents + ZeroResource with one parent Absorbed```
In part because it hasn't worked like that before. 

Extract acts similarly in my mind. If we have fully extracted an old resource into one new resource, then the table should reflect one new Resource, i.e.
```Resource => NewResource with parent Resource and the same mass.```

As far as I can tell, Cyclus does not already record zero resources anywhere else, so if we did here we would be changing the overall logic of the Resource table. which I'm trying to avoid doing because I don't think doing so is on my critical path

Reply:
Wasn't this happening before #1760 when all of the quantity was extracted from an initial material object?

Reply:
I guess yes technically but I don't think any archetype was actually extracting a full resource (there's really no reason to, if a resource is the right size you would just pop it and use as is)

Reply:
I guess a separate and related question is whether depleted resources formally "expire" or just linger...  that may depend on the archetypes they experience, I guess.

Reply:
> I guess yes technically but I don't think any archetype was actually extracting a full resource (there's really no reason to, if a resource is the right size you would just pop it and use as is)

Not all resources are in buffers.  I could imagine an algorithm that calls `ExtractRes()` on one resources and happens to extract all of the first one. i.e. it doesn't bother checking to see whether they're the same size.

Reply:
> As far as I can tell, Cyclus does not already record zero resources anywhere else, so if we did here we would be changing the overall logic of the Resource table. which I'm trying to avoid doing because I don't think doing so is on my critical path

I agree that this is not in scope for you.  I think this all actually depends on many aspects of how a resource is managed in and out of ResBufs.  In many cases, they may ultimately just be merged with other resources through arbitrary operations that are properly tracked by the tracker.

I'm worried about an edge case where some algorithm chooses to neglect a resource whose quantity is too small, and we effectively lose track of the resource.

Reply:
Let's say that Material M is at the front of a ResBuf and some packaging operation results in it having `0 < M.quantity < eps_rsrc()`.  It will still be at the front of the ResBuf.  When the next pacaking request comes, we'll pop M and things behind it, and merge them all into a single Material M'.  That new material will have M as one of it's parents, but in the state it was in before the first round of packaging happened.